revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+    """
+    def moveOneMonthForward(self, date):
+        new_month = (date.month)%12 + 1
+        if new_month==1:
+            new_year = date.year + 1
+            new_date = date.replace(month=new_month, year=new_year)
+        else:
+            new_date = date.replace(month=new_month)
+        return new_date
+
+    """
+    Sets up subdictionaries of each Payment object model for a certain date.
+
+    :args:
+        filtered_dict: The JSON serializable dictionary to be modified.
+        date: The current date period we are iterating over.
+    """
+    def makePaymentTypeKeys(self, filtered_dict, date):
+        filtered_dict['cash_in']['repayments'][date] = {}
+        filtered_dict['cash_in']['donations'][date] = {}
+        filtered_dict['cash_in']['adjustments'][date] = {}
+        filtered_dict['cash_out']['adjustments'][date] = {}
+        filtered_dict['cash_out']['reinvestments'][date] = {}
+        filtered_dict['cash_balances'][date] = {}
+    
+    """
+    Sets up subdictionaries of each Payment object model for a certain date and project.
+
+    :args:
+        filtered_dict: The JSON serializable dictionary to be modified.
+        date: The current date period we are iterating over.
+        title: The title of the project.
+    """
+    def makeProjectKeys(self, filtered_dict, date, title):
+        filtered_dict['cash_in']['repayments'][date][title] = {}
+        filtered_dict['cash_in']['donations'][date][title] = {}
+        filtered_dict['cash_out']['reinvestments'][date][title] = {}
+    
+    """
+    For a specific date period, adds beginning cash balance, change
+    in cash balance, and final cash balance to the JSON serializable
+    dictionary.
+
+    :args:
+        cash: A subdictionary of the JSON serializable dictionary filtered_dict.
+        cash_balance: The cash balance at the beginning of this period.
+        net_cash_in: Net inflow of cash during this period.
+        net_cash_out: Net outflow of cash during this period.
+    """
+    def setCashBalances(self, cash, cash_balance, net_cash_in, net_cash_out):
+        cash['beginning_cash_balance'] = cash_balance
+        cash['change_in_cash'] = net_cash_in - net_cash_out
+        cash['final_cash_balance'] = cash_balance + net_cash_in - net_cash_out
+        cash['net_cash_in'] = net_cash_in
+        cash['net_cash_out'] = net_cash_out
+    
+    """
+    Creates and fills in the 'adjustment' subdictionary of 'cash_in' and 'cash_out'
+    within the JSON serializable dictionary.
+
+    :args:
+        cash: A subdictionary of the JSON serializable dictionary filtered_dict.
+        cash_balance: The cash balance at the beginning of this period.
+        net_cash_in: Net inflow of cash during this period.
+        net_cash_out: Net outflow of cash during this period.
+    """
+    def setTimePeriodAdjustments(self, filtered_dict, date, temp_start_date, temp_end_date):
```

Comment:
This function should definitely be more heavily commented, it's very tough to tell what's going on